clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupController.groovy

```diff
@@ -160,9 +160,35 @@ class ServerGroupController {
   @PostFilter("hasPermission(filterObject?.application, 'APPLICATION', 'READ')")
   @PostAuthorize("@authorizationSupport.filterForAccounts(returnObject)")
   @RequestMapping(value = "/serverGroups", method = RequestMethod.GET)
-  List getServerGroupsByApplications(@RequestParam(value = 'applications') List<String> applications,
-                                     @RequestParam(required = false, value = 'cloudProvider') String cloudProvider) {
-    applications.collectMany { summaryList(it, cloudProvider) }
+  List getServerGroups(@RequestParam(required = false, value = 'applications') List<String> applications,
+                       @RequestParam(required = false, value = 'serverGroupNames') List<String> serverGroupNames,
+                       @RequestParam(required = false, value = 'cloudProvider') String cloudProvider) {
+    if ((applications && serverGroupNames) || (!applications && !serverGroupNames)) {
+      throw new IllegalArgumentException("Provide either 'applications' or 'serverGroupNames' parameter, but not both");
+    }
+
+    if (applications) {
+      return getServerGroupsForApplications(applications, cloudProvider)
+    } else {
+      return getServerGroupsForNames(serverGroupNames)
+    }
+  }
+
+  private List<ServerGroupViewModel> getServerGroupsForApplications(List<String> applications, String cloudProvider) {
+    return applications.collectMany { summaryList(it, cloudProvider) }
+  }
+
+  private List<ServerGroupViewModel> getServerGroupsForNames(List<String> serverGroupNames) {
+    return serverGroupNames.collect {
+      def segments = it.split(':')
+      if (segments.size() != 3) {
+        throw new IllegalArgumentException("Expected serverGroupNames in the format [account1:region1:name1,account2:region2:name2] but got " + serverGroupNames)
```

Comment:
Maybe move this outside the loop and split every `serverGroupName`, and throw an exception containing only the ones that failed to `split`.

Error message is also misleading, at this point you're not dealing with a query parameter so no need to comma-separate ... might be better as some variant of:

```Expected server group names like <account>:<region>:<name> but got ${invalidServerGroupNames.join(", ")```